src/groovy/org/transmartproject/rest/protobuf/HighDimBuilder.groovy

```diff
@@ -108,62 +159,35 @@ class HighDimBuilder {
         Number.isAssignableFrom(clazz) ? ColumnType.DOUBLE : ColumnType.STRING
     }
 
-    public static Map<String, Class> getDataProperties(Projection projection) {
+    private SortedMap<String, Class> getDataProperties(Projection projection) {
         if (projection instanceof MultiValueProjection) {
             MultiValueProjection mvp = projection as MultiValueProjection
-            return mvp.dataProperties
+            Map<String, Class> dataProperties = Maps.transformValues(
+                    mvp.dataProperties,
+                    HighDimBuilder.&decideColumnValueType as Function)
+            ImmutableSortedMap.copyOf dataProperties
         } else {
-            return [('value'): Double]
+            // find the type of the first non-value of the first row
+            ImmutableSortedMap.of 'value',
+                    decideColumnValueType(
+                            rowsIterator.peek().find().getClass())
```

Comment:
There is no "current" element at this point, as the iterator has not been advanced (and even if there was, `Iterator` has `current()` method). `peek()` is basically `next()` but without advancing the iterator.